### PR TITLE
Backport CONFIGURABLE_WASM_LIMITS from 2.1

### PR DESCRIPTION
**Pull request by swatanabe**

### CODE REVIEW

**Comment by spoonincode** [comment]
Is it a requirement that this implementation match that as the 2.1 protocol feature? If not, it seems worthwhile to define a new protocol feature instead so that the fix for EOSIO/eos#10773 can be integrated

**Comment by tbfleming** [comment]
> Is it a requirement that this implementation match that as the 2.1 protocol feature?

It's not

**Comment by swatanabe** [comment]
Yeah, I'll add that fix.  There's another issue I'd like to fix as well (large custom sections can setcode but cannot execute), which wasn't really worth making a new protocol feature for.

**Comment by tbfleming** [comment]
@swatanabe is it ok for me to merge this with my PR? I still need to add snapshot support and you've already ported most of what I need.

File: unittests/test-contracts/wasm_config_bios/wasm_config_bios.cpp [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#include <eosio/contract.hpp>
+
+extern "C" __attribute__((eosio_wasm_import)) void set_wasm_parameters_packed(const void*, std::size_t);
+#ifdef USE_EOSIO_CDT_1_7_X
```
Comment (tbfleming) [inline unittests/test-contracts/wasm_config_bios/wasm_config_bios.cpp:4]: Remove the ifdef; CDT 1.6.3 needs the imports

**Comment by swatanabe** [comment]
> @swatanabe is it ok for me to merge this with my PR? I still need to add snapshot support and you've already ported most of what I need.

Yes.  I don't expect any further changes to conflict with what you're doing.

**Review by tbfleming — APPROVED** [review]

**Comment by tbfleming** [comment]
While reviewing diffs for this I noticed that I missed bringing over the `entry_db` and `reset_memory_fail ` tests when I backported eos-vm. I'll create a PR for that this week.

**Pull request merged**